salt/minion.py

```diff
@@ -585,9 +585,10 @@ def __init__(self, opts):
         # Clean out the proc directory (default /var/cache/salt/minion/proc)
         if (self.opts.get('file_client', 'remote') == 'remote'
                 or self.opts.get('use_master_when_local', False)):
-            if HAS_ZMQ:
-                zmq.eventloop.ioloop.install()
-            io_loop = LOOP_CLASS.current()
+            if self.opts['transport'] == 'zeromq' and HAS_ZMQ:
+                io_loop = zmq.eventloop.ioloop.ZMQIOLoop()
+            else:
+                io_loop = LOOP_CLASS.current()
```

Comment:
A more clear depiction of this logic would be:

```
            if self.opts['transport'] == 'zeromq':
                io_loop = LOOP_CLASS()
            else:
                io_loop = LOOP_CLASS.current()
```

Since we know if the transport if ZeroMQ, HAS_ZMQ must be true (otherwise it would fail to load at the transport level). But you are right to question whether or not using `current()` is appropriate for the other transports. If not, then the logic may go back to a one-liner.